vortex-array/src/array/varbin/compute/filter.rs

```diff
@@ -59,12 +65,12 @@ where
     usize: AsPrimitive<O>,
 {
     let logical_validity = validity.to_logical(offsets.len() - 1)?;
-    if let Some(val) = logical_validity.to_null_buffer()? {
+    if let AllOr::Some(validity) = logical_validity.boolean_buffer() {
         let mut builder = VarBinBuilder::<O>::with_capacity(selection_count);
 
-        for (start, end) in mask.slices().iter().copied() {
-            let null_sl = val.slice(start, end - start);
-            if null_sl.null_count() == 0 {
+        for (start, end) in mask_slices.iter().copied() {
+            let null_sl = validity.slice(start, end - start);
+            if null_sl.count_set_bits() == 0 {
```

Comment:
this is an incorrect translation, fixed in #2144. Null count is not the same thing as count_set_bits